MREUnityRuntime/MREUnityRuntimeLib/App/MixedRealityExtensionApp.cs

```diff
@@ -1027,17 +1027,25 @@ private void OnSetSoundState(SetSoundState payload)
                     var audioClip = MREAPI.AppsAPI.AssetCache.GetAsset(payload.SoundAssetId) as AudioClip;
                     if (audioClip != null)
                     {
-                        var soundInstance = actor.gameObject.AddComponent<AudioSource>();
-                        soundInstance.clip = audioClip;
-                        soundInstance.time = payload.StartTimeOffset;
-                        soundInstance.spatialBlend = 1.0f;
-                        soundInstance.spread = 90.0f;   //only affects multichannel sounds. Default to 50% spread, 50% stereo.
-                        soundInstance.minDistance = 1.0f;
-                        soundInstance.maxDistance = 1000000.0f;
-                        ApplySoundStateOptions(soundInstance, payload.Options);
-                        soundInstance.Play();
-                        _soundInstances.Add(payload.Id, soundInstance);
-                        _unpausedSoundInstances.Add(payload.Id);
+                        float offset = payload.StartTimeOffset;
+                        if (payload.Options.Looping != null && payload.Options.Looping.Value)
+                        {
+                            offset = payload.StartTimeOffset%audioClip.length;
```

Comment:
Please leave spaces between operators.  Makes this difficult to read when looking through code quickly.